test/integration/feeCenter.js

```diff
@@ -0,0 +1,159 @@
+        helpers.api.listed(res, 4);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /fee_centers?available=1&detailed=1 returns a detailed list of available fee centers', function () {
+    return agent.get('/fee_centers?available=1&detailed=1')
+      .then(function (res) {
+        helpers.api.listed(res, 1);
+      })
+     .catch(helpers.handler);
+  });
+
+  it('GET /fee_centers?is_cost=1&is_principal=1 returns a list of principal fee center which are cost', function () {
+    return agent.get('/fee_centers?is_cost=1&is_principal=1')
+      .then(function (res) {
+        helpers.api.listed(res, 3);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /fee_center/:id returns one fee center', function () {
+    return agent.get('/fee_centers/'+ FETCHABLE_FEE_CENTER_ID)
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(res.body).to.not.be.empty;
+        expect(res.body.id).to.be.equal(FETCHABLE_FEE_CENTER_ID);
+        expect(res.body).to.have.all.keys(responseKeys);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /fee_center/:id returns a 404 for an unknown fee center id', function () {
+    return agent.get('/fee_centers/unknown')
+      .then(function (res) {
+        helpers.api.errored(res, 404);
+      })
+      .catch(helpers.handler);
+  });
+
+
+
+  /* @todo - should this return a simple number? */
+  it('GET /fee_centers/:id/value returns the value of a provided fee center', function () {
+    return agent.get(`/fee_centers/${FETCHABLE_FEE_CENTER_ID}/value`)
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(res.body).to.not.be.empty;
+        expect(res.body).to.have.all.keys('id', 'value');
+        expect(res.body.value).to.satisfy(function (value) { return value >= 0;});
+      })
+      .catch(helpers.handler);
+  });
+
+  it('POST /fee_centers adds a fee center', function () {
+    return agent.post('/fee_centers')
+      .send(newFeeCenter)
+      .then(function (res) {
+        helpers.api.created(res);
+        newFeeCenter.id = res.body.id;
+        return agent.get('/fee_centers/' + newFeeCenter.id);
+      })
+      .then(function (res){
+        expect(res).to.have.status(200);
+        expect(res.body).to.have.all.keys(responseKeys);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('PUT /fee_centers/:id updates the newly added fee center', function () {
+    var updateInfo = { note : 'update value for note' };
+    return agent.put('/fee_centers/' + newFeeCenter.id)
+      .send(updateInfo)
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(res.body.id).to.equal(newFeeCenter.id);
+        expect(res.body.note).to.equal(updateInfo.note);
+      })
+      .catch(helpers.handler);
+  });
+
+  it.skip('DELETE /fee_centers/:id deletes a fee_center', function () {
```

Comment:
We did not plan a deletion operation for fee center so they are there just in case there is a different decision.